src/main/java/com/conveyal/analysis/components/HttpApi.java

```diff
@@ -167,4 +196,9 @@ public HttpApi (
         return sparkService;
     }
 
+    // Maybe this should be done or called with a JVM shutdown hook
```

Comment:
Are there other services that we should add to the shutdown hook? AWS connections or the MongoDB driver?

Reply:
Yes, I think we'll want to move toward cleanly shutting down MapDB, HTTP servers, AWS connections, MongoDB, everything in a JVM shutdown hook. I'll make a ticket for that.